pkg/ibctypes/validator_set.go

```diff
@@ -0,0 +1,127 @@
+// Package ibctypes defines to handle IBC handshakes in SPN modules
+package ibctypes
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strconv"
+
+	ibctmtypes "github.com/cosmos/ibc-go/modules/light-clients/07-tendermint/types"
+	"github.com/pkg/errors"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"gopkg.in/yaml.v2"
+)
+
+const TypeEd25519 = "tendermint/PubKeyEd25519"
+
+// ParseValidatorSetFromFile parses a YAML dumped Validator Set file and returns a new Validator Set
+// TODO: Support other format than YAML if there are other format of dumped file
+func ParseValidatorSetFromFile(filePath string) (vs ValidatorSet, err error) {
+	// parse file
+	var vsf struct {
```

Comment:
maybe we should create a global struct type outside the method for it

Reply:
Yes, we also track this part for refactoring. Since the logic is off-chain, this is lower priority